addons/udes_stock/tests/test_picking_batch.py

```diff
@@ -804,24 +804,142 @@ def test27_unpickable_item_product_ok_two_picks(self):
+        self.assertEqual(picking.state, 'confirmed',
+                         'picking was not confirmed')
+        # Check no backorder has been created
+        self.assertEqual(len(picking.u_created_back_orders), 0)
+        # check that the investigation has reserved 3 only
+        inv_picking = Picking.search([('picking_type_id', '=', self.picking_type_internal.id)])
+        self.assertEqual(len(inv_picking), 1)
+        self.assertEqual(len(inv_picking.move_line_ids), 1)
+        self.assertEqual(inv_picking.move_line_ids[0].product_qty, 3)
+
+    def test28_unpickable_item_product_ok_in_two_picks(self):
+        """
+        Tests that calling unpickable item for a product with location
+        ends up with all the quant reserved for the stock investigation
+        and the picking remains in state confirmed since there is no
+        more stock. In case the quant is reserved for more than one picking
+        the stock investigation will contain only the quantity of the
+        unpickable + available quantity of the quant.
+        Example: quant of 4, 1 unit reserved in two pickings, both of which are
+                 in the batch, leaves an available quantity of 2, so when
+                 unpickable of one of the pickings will create an investigation of 4.
+        """
+        Picking = self.env['stock.picking']
+        Batch = self.env['stock.picking.batch']
+        Batch = Batch.sudo(self.outbound_user)
+
+        quant = self.create_quant(self.apple.id, self.test_location_01.id, 4)
+
+        products_info = [{'product': self.apple, 'qty': 1}]
+        picking = self.create_picking(self.picking_type_pick,
                                       products_info=products_info,
                                       confirm=True,
                                       assign=True)
+
+        picking_2 = self.create_picking(self.picking_type_pick,
+                                        products_info=products_info,
+                                        confirm=True,
+                                        assign=True)
         # check that now 2 units of the quant are reserved
         self.assertEqual(quant.reserved_quantity, 2)
 
         batch = Batch.create_batch(self.picking_type_pick.id, None)
 
         self.assertIn(picking, batch.picking_ids)
+        batch.add_extra_pickings(self.picking_type_pick.id)
+        self.assertIn(picking_2, batch.picking_ids)
 
         reason = 'missing item'
         move_line = picking.move_line_ids[0]
 
         batch.unpickable_item(product_id=move_line.product_id.id,
                               location_id=move_line.location_id.id,
                               reason=reason)
-        # after unickable all the quant should be reserved
+        # after unpickable all the quant should be reserved
+        self.assertEqual(quant.reserved_quantity, 4)
+        # picking state should be confirmed
+        self.assertEqual(picking.state, 'confirmed',
```

Comment:
Do we want to check this for both pickings? (Same for line 886)

Reply:
Done